### PR TITLE
Story7 sc

### PR DESCRIPTION
**Pull request by CarlOllington**

### CODE REVIEW

File: product.php [hunk 1]
```diff
@@ -8,41 +8,56 @@
...
+            {
+                CurrencyConversion::setCurrency($currencyRequest);
+                $response = Response::apiResponse(200,'Successfully retrieved products',
+                    ProductsHydrator::getProduct(Connection::db(), intval($_GET['id'])));
+            } else
+            {
+                throw new InvalidCurrencyException();
+            }
             $response = Response::apiResponse(200, 'Successfully retrieved product',
                 ProductsHydrator::getProduct(Connection::db(), intval($_GET['id'])));
```
Comment (ramsey-merdassi) [inline product.php:46]: Duplicated above in currency

Comment (CarlOllington) [reply product.php:46]: implemented.

File: src/Products/Products.php [hunk 2]
```diff
@@ -16,7 +17,7 @@ public function jsonSerialize(): mixed
     {
         return [
             'id' => $this->id,
-            'price' => $this->price,
+            'price' => CurrencyConversionService::ConvertCurrency($this->price),
```
[the remark this answers is not in the document]
Comment (CarlOllington) [reply src/Products/Products.php:20]: implemented.

File: src/Services/CurrencyConversion.php [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        {
+            $output = round($price * 1.16, 2);
+        }
+        elseif (self::$currency === 'YEN')
+        {
+            $output = round($price * 162.16, 2);
+        }
+        else
+        {
+            $output = $price;
```
[the remark this answers is not in the document]
Comment (CarlOllington) [reply src/Services/CurrencyConversion.php:30]: implemented. matches unit class.

**Review by matt-ioa — APPROVED** [review]

**Pull request merged**